tests/unit/net/fetch.rs

```diff
@@ -213,15 +215,19 @@ fn test_cors_preflight_fetch() {
     };
     let (mut server, url) = make_server(handler);
 
+    let target_url = url.clone().join("a.html").unwrap();
+
     let origin = Origin::Origin(UrlOrigin::new_opaque());
-    let mut request = Request::new(url, Some(origin), false, None);
-    *request.referer.borrow_mut() = Referer::NoReferer;
+    let mut request = Request::new(url.clone(), Some(origin), false, None);
+    *request.referer.borrow_mut() = Referer::RefererUrl(target_url);
+    *request.referrer_policy.get_mut() = Some(ReferrerPolicy::Origin);
     request.use_cors_preflight = true;
     request.mode = RequestMode::CORSMode;
     let fetch_response = fetch_sync(request, None);
     let _ = server.close();
 
     assert!(!fetch_response.is_network_error());
+    assert_eq!(url, fetch_response.url.unwrap());
```

Comment:
Oh right, because we're still using `url` for the target of the Request. Too many URLs to keep track of!